### PR TITLE
feat: dynamic generation mode

### PR DESCRIPTION
**Pull request by JimmyPaolini**
Adding a configuration option `dynamic?: boolean` that defaults to false and makes no changes to the output, but when set to true generates mock values dynamically at run time (when a mock function is called), rather than statically at build time (when a mock function is generated).

https://github.com/ardeois/graphql-codegen-typescript-mock-data/issues/85

### CODE REVIEW

**Comment by JimmyPaolini** [comment]
@ardeois Let me know what you think of this solution for issue #85!

File: src/index.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ type Options<T = TypeNode> = {
     currentType: T;
     customScalars?: ScalarMap;
     transformUnderscore: boolean;
+    dynamic?: boolean;
```
Comment (ardeois) [inline src/index.ts:24]: could you name this `dynamicValues` maybe?

File: tests/typescript-mock-data.spec.ts [hunk 2]
```diff
@@ -389,3 +389,10 @@ it('should preserve underscores if transformUnderscore is false', async () => {
     );
     expect(result).toMatchSnapshot();
 });
+
+it('should generate dynamic mocks', async () => {
+    const result = await plugin(testSchema, [], { dynamic: true });
+
+    expect(result).toBeDefined();
+    expect(result).toMatchSnapshot();
+});
```
Comment (ardeois) [inline tests/typescript-mock-data.spec.ts:398]: You should probably create another unit test to make sure the `casual.seed` works in unit test

For example, create a new directory like in `tests/circular-mocks`, generate the mocks with a new `create-mocks.ts` file that you run [in jest](https://github.com/ardeois/graphql-codegen-typescript-mock-data/blob/86dbdf1cfd566074637c7119164175e71d3fbd46/jest.config.js#L6) and create a unit test with a call to `casual.seed` like in [here](https://github.com/ardeois/graphql-codegen-typescript-mock-data/blob/main/tests/circular-mocks/spec.ts)

This way, we'll be able to check if the unit test always returns the same values.
You can have a test like:
```typescript
    const a = anA();
    expect(a).toEqual({ foo: 'value' });
```

Where `value` is a string generated by `casual.word` which is called at every test run
If the `casual.seed` works, then the unit test should return the same value every time

**Review by ardeois — COMMENTED** [review]
Looks good ! Thanks for your contribution !

I have a few comments before we can go forward with this. Sorry for the delay, I was quite busy these last few days

File: src/index.ts (hunk 1 above)
Comment (JimmyPaolini) [reply src/index.ts:24]: Sure!

File: tests/terminateCircularRelationships/mocks.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { A, B, C, D } from './types';
```
Comment (ardeois) [inline tests/terminateCircularRelationships/mocks.ts:1]: This file should not be commited as it's generated by `globalSetup.ts`
You can add it to the `.gitignore` like [here](https://github.com/ardeois/graphql-codegen-typescript-mock-data/blob/main/.gitignore#L18)

And you can remove the special entry you added in `.eslintignore`

**Review by ardeois — COMMENTED** [review]
Thank you for the udpates, all the code is good now I think

One last thing I forgot, I think it's important to update the readme to explain how to use it
I would suggest to update the [configuration](https://github.com/ardeois/graphql-codegen-typescript-mock-data#configuration) section, and link to the dynamicValues unit test so developers can see an example of usage

File: tests/terminateCircularRelationships/mocks.ts (hunk 3 above)
Comment (JimmyPaolini) [reply tests/terminateCircularRelationships/mocks.ts:1]: Great!

File: README.md [hunk 4]
```diff
@@ -129,6 +129,10 @@ export const aUser = (overrides?: Partial<User>): User => {
 
 When disabled, underscores will be retained for type names when the case is changed. It has no effect if `typenames` is set to `keep`.
 
+### dynamicValues (`boolean`, defaultValue: `false`)
+
+When enabled, values will be generated dynamically when the mock function is called rather than statically when the mock function is generated. The values are generated consistently from a [casual seed](https://github.com/boo1ean/casual#seeding) that can be manually configured using the generated `seedMocks(seed: number)` function, as shown in [this test](https://github.com/JimmyPaolini/graphql-codegen-typescript-mock-data/blob/dynamic-mode/tests/dynamicValues/spec.ts#L13).
```
Comment (ardeois) [inline README.md:134]: 👍

**Review by ardeois — APPROVED** [review]
Looks good !
Thanks for all the updates, I'll merge the PR

**Pull request merged**